### PR TITLE
Bug fix for webapp ssh

### LINKED ISSUE #8281: [webapp ssh] sometimes the connection refused
I ran into connection refused issue several time when tried to connect but was able to connect after couple of tries. We should look into the issue, you may be able to get repro logs from https://petclinic.azurewebsites.net/ 
```
yiliao@yili-ubuntu-03:~$ az webapp ssh -g appsvc_rg_Linux_centralus -n yili-python-up-08
Auto-selecting port: 34229
SSH is available ( username: root, password: Docker! )
_ssh: connect to host localhost port 34229: Connection refused_
```

### LINKED ISSUE #8282: [webapp ssh] fails after 20mins of idle
, after idle for more than 20 minutes (maybe?), I got this: 
……………………………………..
OSError: [Errno 9] Bad file descriptor
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/tunnel.py", line 168, in _listen_to_client
    nbytes = client.recv_into(buf, 4096)
OSError: [Errno 9] Bad file descriptor
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/tunnel.py", line 168, in _listen_to_client
    nbytes = client.recv_into(buf, 4096)
Fatal Python error: could not acquire lock for <_io.BufferedWriter name='<stdout>'> at interpreter shutdown, possibly due to daemon threads

Thread 0x00007fd195388700 (most recent call first):
  File "/opt/az/lib/python3.6/traceback.py", line 101 in print_exception
  File "/opt/az/lib/python3.6/traceback.py", line 159 in print_exc
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/tunnel.py", line 181 in _listen_to_client
  File "/opt/az/lib/python3.6/threading.py", line 864 in run
  File "/opt/az/lib/python3.6/threading.py", line 916 in _bootstrap_inner
  File "/opt/az/lib/python3.6/threading.py", line 884 in _bootstrap

Thread 0x00007fd19638a700 (most recent call first):
  File "/opt/az/lib/python3.6/threading.py", line 1072 in _wait_for_tstate_lock
  File "/opt/az/lib/python3.6/threading.py", line 1056 in join
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/tunnel.py", line 137 in _listen
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/tunnel.py", line 186 in start_server
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2296 in _start_tunnel
  File "/opt/az/lib/python3.6/threading.py", line 864 in run
  File "/opt/az/lib/python3.6/threading.py", line 916 in _bootstrap_inner
  File "/opt/az/lib/python3.6/threading.py", line 884 in _bootstrap

Current thread 0x00007fd19eaad700 (most recent call first):
/usr/bin/az: line 2: 30043 Aborted                 (core dumped) /opt/az/bin/python3 -Im azure.cli "$@"

### PR DESCRIPTION
**Pull request by patricklee2**
fixes #8282  #8281, #8280 , #8279

uses fabric ssh library which as a BSD 2-clause license
Added license at  src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/fabric_license
im not sure if that is the correct place top out it.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by patricklee2** [comment]
@yugangw-msft @panchagnula @yiliaomsft @rramachand21 
This is ready for review

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2281,46 +2283,69 @@ def create_tunnel(cmd, resource_group_name, name, port=None, slot=None):
...
+    while True:
+        try:
+            c = Connection(host=hostname,
+                           port=port,
+                           user=username,
+                           # connect_timeout=60*10,
+                           connect_kwargs={"password": password})
+            break
+        except Exception:  # pylint: disable=broad-except
+            if tries == 0:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: I would suggest you log an verbose entry to document the exception details: `logger.info(ex)`. My concern is catching  generic exceptions above could swallow non connection related errors which ends up making the diagnosis pretty hard.

File: src/command_modules/azure-cli-appservice/setup.py [hunk 2]
```diff
@@ -38,6 +38,8 @@
     # v1.17 breaks on wildcard cert https://github.com/shazow/urllib3/issues/981
     'urllib3[secure]>=1.18',
     'xmltodict',
+    'fabric>=2.4',
+    'cryptography==2.3.1',
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/setup.py]: `cryptography` is a pretty common dependency, why we need to ping to a specific version?

Comment (patricklee2) [reply src/command_modules/azure-cli-appservice/setup.py]: paramiko, (one of the dependency) is still using a deprecated method.
as a result, these warnings appear, and I couldn't suppress them
the fix was to pin the version.

```/home/patle/azure-cli/env/lib/python3.7/site-packages/paramiko/kex_ecdh_nist.py:39: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
  m.add_string(self.Q_C.public_numbers().encode_point())
/home/patle/azure-cli/env/lib/python3.7/site-packages/paramiko/kex_ecdh_nist.py:96: CryptographyDeprecationWarning: Support for unsafe construction of public numbers from encoded data will be removed in a future version. Please use EllipticCurvePublicKey.from_encoded_point
  self.curve, Q_S_bytes
/home/patle/azure-cli/env/lib/python3.7/site-packages/paramiko/kex_ecdh_nist.py:111: CryptographyDeprecationWarning: encode_point has been deprecated on EllipticCurvePublicNumbers and will be removed in a future version. Please use EllipticCurvePublicKey.public_bytes to obtain both compressed and uncompressed point encoding.
  hm.add_string(self.Q_C.public_numbers().encode_point())

```

Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/setup.py]: would be reasonable to have "<2.5.0"?

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (patricklee2) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: added

File: src/command_modules/azure-cli-appservice/setup.py (hunk 2 above)
Comment (patricklee2) [reply src/command_modules/azure-cli-appservice/setup.py]: changed it to <2.5.0

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -2281,46 +2283,70 @@ def create_tunnel(cmd, resource_group_name, name, port=None, slot=None):
...
+                logger.warning('Connection is not ready yet, please wait')
+            if tries == 60:
+                raise CLIError("Timeout Error, Unable to establish a connection")
+            tries = tries + 1
+            logger.warning('.')
+            time.sleep(1)
+    try:
+        c.run('cat /etc/motd', pty=True)
+        c.run('source /etc/profile; /bin/ash', pty=True)
```
Comment (rramachand21) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: this is supposed to be /bin/bash right?

File: src/command_modules/azure-cli-appservice/setup.py (hunk 2 above)
Comment (ctaggart) [reply src/command_modules/azure-cli-appservice/setup.py]: The `<2.5.0` now causes a conflict https://github.com/Azure/azure-cli/issues/9629. Please update.

**Pull request merged**